soundzones_protocol/documentation_SZP.md

```diff
@@ -0,0 +1,257 @@
+
+# Documentation for SoundZone Protocol
+<!-- 
+To compile puml use: (Assuming plantuml you are in the directory)
+plantuml.jar -tsvg readme.md -o sequence_diagrams
+-->
+
+version = 0.0.2
+
+This file will document SZP.
+
+As it is a 7th layer protocol, it is treated as a Point-Point communication
+
+| Defines | Values |
+|---|---|
+| IPv | IPv4 |
+| Transport protocol | UDP |
+| Port | 1695 ( Seemingly not used ) |
+| Transmission encoding | utf-8 |
+
+## Main Structure
+
+[CID, {payload} ]
+
+* CID - Command Id
+* payload - Data to send (Structure depends on the CID)
+
+## Commands 
+
+### Command groups
+
+| Range | Name |
+|---| ---|
+| 0x00 - 0x10 | Normal use |
+| 0xA0 - 0xC0 | Configuration |
+| 0xF0 - 0xFF | Debugging |
+
+### Command list
+
+|CID | Name | Description |
+|---|---|---|
+| 0x01 | send | Send a payload to a client |
+|||
+| 0xA1 | enroll_c | Used by client to enroll |
+| 0xA2 | enroll_m | Used by client to enroll |
+| 0xB1 | set_time_encode | Sets the encoding of the time |
+| 0xB2 | set_size_payload_len | Sets size of payload_len |
+| B3 | set_sound_compression | Sets the compression for the music |
+| B4 | set_sample_rate | Sets the sample rate of the music |
+|||
+| 0xF1 | checkCon | Used to check connection, used for debug |
+
+---
+
+### 01 - send
+
+To use this command, client has been preconfigured with a Id.
+So the rigth filters is used for the right Id.
+
+| Tag | Size [bytes] | Value | Description | 
+|---|---|---|---|
+| cid | 1 | 0x01 | Command Id |
+| time | Depends on encoding | - | Time to play the block of sound |
+| payload_len | Depends on encoding | - | Length of payload |
+| payload | 1 | - | Payload |
+
+<!--
+```
+@startuml 01_send
+server -> client: [ cid, time, payload_len, payload ]
+@enduml
+```
+-->
+
+![](sequence_diagrams/01_send.svg)
+
+
+---
+
+### A1 - enroll_c - client assigned id's
+To use this command, client has been preconfigured with a Id.
+So the rigth filters is used for the right Id.
+
+| Tag | Size [bytes] | Value | Description | 
+|---|---|---|---|
+| cid | 1 | 0xA1 | Command Id |
+|  client_id | 1 | 0x01-0xFF | Assigned client id (0 is reserved for server) |
+| res | 1 | 0 \| 1 | denie \| accept |
+
+<!--
+```
+@startuml A1_enrole_c
+server <- client: [ cid, client_id ]
+server -> client: [ cid, res ]
+@enduml
+```
+-->
+
+![](sequence_diagrams/A1_enrole_c.svg)
+
```

Comment:
Hvis client er preconfigureret med et ID, hvad skal den her commando så bruges til.  Er den kun for at bekræfte at 2 ikke har fået det samme iD?